Sitefinity CLI/PackageManagement/NuGetApiClient.cs

```diff
@@ -33,26 +33,54 @@ public async Task<NuGetPackage> GetPackageWithFullDependencyTree(string id, stri
             if (File.Exists(packageDependenciesHashFilePath))
                 return JsonConvert.DeserializeObject<NuGetPackage>(File.ReadAllText(packageDependenciesHashFilePath));
 
-            XDocument nuGetPackageXmlDoc = await this.GetPackageXmlDocument(id, version, sources);
+            this.isV2 = true;
+            XDocument nuGetPackageXmlDoc = await this.GetPackageXmlDocument(id, version, sources, httpClient);
             if (nuGetPackageXmlDoc == null)
             {
                 return null;
             }
 
-            XElement propertiesElement = nuGetPackageXmlDoc
-                .Element(this.xmlns + Constants.EntryElem)
-                .Element(this.xmlnsm + Constants.PropertiesElem);
-
+            IList<NuGetPackage> dependencies = new List<NuGetPackage>();
             NuGetPackage nuGetPackage = new NuGetPackage();
-            nuGetPackage.Id = nuGetPackageXmlDoc
-                .Element(this.xmlns + Constants.EntryElem)
-                .Element(this.xmlns + Constants.TitleElem).Value;
+            if (this.isV2)
```

Comment:
Encapsulate the logic for v3 and v2 into separate methods for better readability.

Reply:
done